hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java

```diff
@@ -126,39 +127,52 @@
 
     abstract ExceptionAction onException(Throwable t);
 
-    public String getServiceName() {
-        return serviceName;
-    }
+    protected abstract Address getTarget();
 
     InternalPartition getPartition() {
         return nodeEngine.getPartitionService().getPartition(partitionId);
     }
 
+    @Override
+    public boolean isLocal() {
+        return true;
+    }
+
     private long getCallTimeout(long callTimeout) {
         if (callTimeout > 0) {
             return callTimeout;
         }
 
-        final long defaultCallTimeout = operationService.getDefaultCallTimeoutMillis();
-        if (op instanceof WaitSupport) {
-            final long waitTimeoutMillis = op.getWaitTimeout();
-            if (waitTimeoutMillis > 0 && waitTimeoutMillis < Long.MAX_VALUE) {
-                /*
-                 * final long minTimeout = Math.min(defaultCallTimeout, MIN_TIMEOUT);
-                 * long callTimeout = Math.min(waitTimeoutMillis, defaultCallTimeout);
-                 * callTimeout = Math.max(a, minTimeout);
-                 * return callTimeout;
-                 *
-                 * Below two lines are shortened version of above*
-                 * using min(max(x,y),z)=max(min(x,z),min(y,z))
-                 */
-                final long max = Math.max(waitTimeoutMillis, MIN_TIMEOUT);
-                return Math.min(max, defaultCallTimeout);
-            }
+        long defaultCallTimeout = operationService.defaultCallTimeoutMillis;
+        if (!(op instanceof WaitSupport)) {
+            return defaultCallTimeout;
+        }
+
+        long waitTimeoutMillis = op.getWaitTimeout();
+        if (waitTimeoutMillis > 0 && waitTimeoutMillis < Long.MAX_VALUE) {
+            /*
+             * final long minTimeout = Math.min(defaultCallTimeout, MIN_TIMEOUT);
+             * long callTimeout = Math.min(waitTimeoutMillis, defaultCallTimeout);
+             * callTimeout = Math.max(a, minTimeout);
```

Comment:
I only cleaned up the structure of the code. So the wait timeout was already there. It isn't changed by me.